pkg/metrics/metrics.go

```diff
@@ -124,9 +125,18 @@ func (sc *samplesCollector) Collect(ch chan<- prometheus.Metric) {
 		skips[skip] = true
 	}
 
-	importFailuresExist := util.ConditionTrue(cfg, configv1.ImportImageErrorsExist)
-	importFailures := util.Condition(cfg, configv1.ImportImageErrorsExist)
-	importFailuresReason := importFailures.Reason
+	cms, err := sc.configmaps.List(labels.Everything())
```

Comment:
Shouldn't we use a label to identify the `ConfigMaps` that contain imagestream errors?

Reply:
updates are often happening rather abundantly/concurrently when errors are going down ... I'm not sure a preliminary filter like that will help us much .... the code is simpler on the other side if we just maintain the k/v for each tag in the configmap and don't add the additional logic to deal with turning the label on and off ... more moving parts IMO

Reply:
I was thinking that we use the same label key/value for these ConfigMaps so we don't have to list every ConfigMap in the namespace. Ex: `samples.openshift.io/imagestream-error: true`

Reply:
yeah I get that piece ... its the added complexity around when/how to set / change the value of the label that I wonder about
and whether we gain much in the end

perhaps I'll see about coding it up so we can decide one way or the other

Reply:
They key point is not just tracking whether errors exist or not, but when we retry imports ... and when we retry imports, it is on a per imagestreatag basis.

Reply:
But looking at the code again now ...

Reply:
... and I'm wondering if I found another needed change looking into this :-)